src/QMCTools/PyscfToQmcpack_Spline.py

```diff
@@ -0,0 +1,947 @@
+######################################################################################
+## This file is distributed under the University of Illinois/NCSA Open Source License.
+## See LICENSE file in top directory for details.
+##
+## Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+##
+## File developed by: Thomas Applencourt, applencourt@anl.gov,  Argonne National Laboratory
+##                    Hyeondeok Shin, hshin@anl.gov, Argonne National Laboratory
+##                    Yubo "Paul" Yang, yubo.paul.yang@gmail.com, University of Illinois Urbana-Champaign
+##                    Anouar Benali, benali@anl.gov, Argonne National Laboratory
+##                  
+## File created by: Thomas Applencourt, applencourt@anl.gov, Argonne National Laboratory
+#######################################################################################
+
+from __future__ import print_function
+import numpy as np
+import h5py
+import os
+
+def pyscf2qmcpackspline(cell,mf,title="Default", kpts=[], kmesh=[],  sp_twist=[]):
+  import sys, re
+
+  Restricted=True
+  PBC=False
+  Gamma=False
+
+  # Python version check
+  if sys.version_info < (3, 0):
+       sys.exit("Python 2.x not supported")
+
+  # FFT mesh check
+  if (cell.mesh.any() % 2) == 0:
+       sys.exit("Even number of FFT mesh not supported")	
+    
+  #Twists generation not yet implemented
+  if len(sp_twist)== 0:
+       sp_twist=[0.0,0.0,0.0]
+
+  val=str(mf)
+  ComputeMode= re.split('[. ]',val)
+
+  SizeMode=len(ComputeMode)
+  for i in range(SizeMode):
+     if ComputeMode[i] in ("UHF","KUHF","UKS"):
+           Restricted=False
+           sys.exit("Unrestricted calculations not supported")
+
+     if ComputeMode[i]=="pbc":
+           PBC = True
+
+
+  if not PBC:
+        sys.exit("Open boundary condition without lattice vectors not supported")
+     
+  if PBC and len(kpts) == 0:
+        #sys.exit("You need to specify explicit the list of K-point (including gamma)")
+        Gamma=True
+
+  if len(kpts)!= 0:
+#     loc_cell,kmesh=get_supercell(cell,kmesh)
+     sys.exit("K-point scf not supported")
+  else:
+     loc_cell=cell
+ 
+  h5_fname = f"{title}.h5"
+  xml_fname = f"{title}.xml"
+
+  tile = [1, 1, 1]
+  tilematrix = [0]*9
+  for dim, val in enumerate(tile):
+    tilematrix[4*dim] = val
+   
+
+  tilematrix_str = " ".join(map(str,tilematrix))
+
+  gvecs, eig_df = save_eigensystem(mf, save=False)
+
+  # generate wave function file
+  # ================================================
+  generate_pwscf_h5(loc_cell,gvecs,eig_df,h5_fname)  
+
+
+  # generate QMCPACK input file
+  # ================================================
+  from lxml import etree
```

Comment:
The lxml package is not a default python package.   The import should happen earlier, and maybe be checked to print a nicer error message.

Reply:
Good catch. Can we not avoid this? It complicates installation, testing, user support etc. further.

Reply:
Honestly, I agree. 

This work was based on the good work of Yang, Paul. I just updated and fixing some bugs on this code but he did the majority of the work.
He was using already using `lxml`, so I decided to follow is steps.
I can try to see how hard it's to go directly from the native python `xml` parser. 

On the other hand, to use this script people need to install PySCF first, so they can just install `lxml` at the time :).   But I understand, for the qmcpack point of view, having one more dependency is tedious.

Reply:
Honestly, I agree.
This work was based on the good work of Yang, Paul. I just updated and fixing some bugs on this code but he did the majority of the work.
He was using already using lxml, so I decided to follow is steps.
I can try to see how hard it's to go directly from the native python xml parser.
On the other hand, to use this script people need to install PySCF first, so they can just install lxml at the time :slightly_smiling_face:. But I understand, for the qmcpack point of view, having one more dependency is tedious.

Reposted: because of bugs

Reply:
Since it is a PySCF dependency it is not too bad.